### PR TITLE
time: add utimes(), lutimes() and futimes() implementation

### PR DESCRIPTION
**Pull request by ziemleszcz**
JIRA: DTR-276

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [x] New test added: (https://github.com/phoenix-rtos/phoenix-rtos-tests/pull/94).
- [x] Tested by hand on: (imx6ull).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

https://github.com/phoenix-rtos/phoenix-rtos-filesystems/pull/62
https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/304

- [x] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [x] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: sys/time.c [hunk 1]
```diff
@@ -14,24 +14,58 @@
...
+
+	msg.type = mtSetAttr;
+	msg.i.attr.oid = oid;
+
+	if (times == NULL)
+		gettimeofday(&now, NULL);
+
+	msg.i.attr.type = atMTime;
+	msg.i.attr.val = times == NULL ? now.tv_sec : times[1].tv_sec;
+	if (((err = msgSend(oid.port, &msg)) < 0) || ((err = msg.o.attr.err) < 0))
```
Comment (agkaminski) [inline sys/time.c:47]: Assignment inside of if condition is against MISRA rule, we are abandoning this code idiom.

**Review by agkaminski — APPROVED** [review]

File: sys/time.c (hunk 1 above)
Comment (nalajcie) [reply sys/time.c]: hey, do You know if this rule also applies to assignments inside `while` conditions? (should probably apply also, so we need to rewrite our `-EINTR` loops in libphoenix?)

Comment (agkaminski) [reply sys/time.c]: Yeah, I do, missed that one during second reading - to be fixed too.

We need to fix all occurrences, for now during natural code evolution, later as part of making static code analysis pass without errors ;)

**Comment by niewim19** [comment]
As we plan to conform to POSIX as much as possible I have one note:
`futimes()` is not specified in POSIX. `futimens()` is. Difference is basically a name as both functions share the same declaration and `futimes()` errno set is subset of `futimens()`\`s.

[futimens documentation](https://pubs.opengroup.org/onlinepubs/9699919799.2018edition/)
[futimes documentation](https://man7.org/linux/man-pages/man3/futimes.3.html)

We can think about aliasing one with another or vice versa. We better have POSIX function that is not 100% compliant rather than no function at all.

**Comment by nalajcie** [comment]
> As we plan to conform to POSIX as much as possible I have one note: `futimes()` is not specified in POSIX. `futimens()` is. Difference is basically a name as both functions share the same declaration and `futimes()` errno set is subset of `futimens()``s.

Thanks for mentioning this - the issue was raised in related kernel PR (https://github.com/phoenix-rtos/phoenix-rtos-kernel/pull/304) and @ziemleszcz was kind enough to make `sys_futimens` syscall (which means we will have POSIX-compatible kernel API enabling us to implement both `futimes` and `futimens` in libphoenix (not necessarily as a part of this PR)).

**Review by agkaminski — APPROVED** [review]

**Pull request merged**